online-bookstore/online-bookstore/src/main/java/com/example/online_bookstore/implementation/BookOrderServiceRepo.java

```diff
@@ -0,0 +1,67 @@
+package com.example.online_bookstore.implementation;
+
+import com.example.online_bookstore.models.BookOrder;
+import com.example.online_bookstore.repository.BookOrderRepo;
+import com.example.online_bookstore.service.BookOrderService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class BookOrderServiceRepo implements BookOrderService {
+
+    @Autowired
+    BookOrderRepo repo;
+
+    @Override
+    public List<BookOrder> findBookOrderAfterDate(Date date) {
+        return repo.findByStartDateAfter(date);
+    }
+
+    @Override
+    public List<BookOrder> findBookOrderBeforeDate(Date date) {
+        return repo.findByStartDateBefore(date);
+    }
+
+    @Override
+    public List<BookOrder> findBookOrderBetweenDates(Date startDate, Date endDate) {
+        return repo.findByStartDateBetween(startDate, endDate);
+    }
+
+    @Override
+    public List<BookOrder> findCompletedBookOrder(Date date) {
+        return repo.findByOrderFinishedDateBeforeAndOrderFinishedDateIsNotNull(date);
+    }
+
+    @Override
+    public boolean existsById(long id) {
+        return repo.existsById(id);
+    }
+
+    @Override
+    public Optional<BookOrder> findById(long id) {
+        return repo.findById(id);
+    }
+
+    @Override
+    public BookOrder create(BookOrder t) {
+        return repo.save(t);
+    }
+
+    @Override
+    public Optional<BookOrder> update(BookOrder t, long id) {
+        if(existsById(id)) {
+            t.setId(id);
+            return Optional.of(repo.save(t));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public void delete(long id) {
+        repo.deleteById(id);
+    }
```

Comment:
You have a lot of methods, most will not be used in bussiness logic. Think of few functionalities that go from Controller->Service->Repo and just do them. No need to implement all of this when it will not be used in client requests.

Reply:
This also applies to other Services. First make controllers and think of functionalities.